source_modelling/sources.py

```diff
@@ -442,8 +474,103 @@ class Fault:
 
     planes: list[Plane]
 
-    @staticmethod
-    def from_corners(fault_corners: np.ndarray) -> "Fault":
+    def __post_init__(self) -> None:
+        """Ensure that planes are ordered along strike end-to-end."""
+        # The tirivial case where the number of planes is one should be ignored
+        if len(self.planes) == 1:
+            return
+
+        # Planes can only have one dip, dip direction, and width.
```

Comment:
This holds for all NSHM faults?

Reply:
It does for everything except the dip, which I should actually remove because the dip direction and width are the only things that really needs to be constant.